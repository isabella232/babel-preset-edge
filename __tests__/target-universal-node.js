import buildPreset from "../src"

import { check, fixtures, titles } from "./core"

describe("Target: Universal: Node Imports", () => {
  const options = buildPreset(null, {
    sourceMaps: false,
    imports: "node"
  })

  fixtures.forEach((fileName, index) => {
    test(titles[index], () => check(fileName, options))
  })
})
