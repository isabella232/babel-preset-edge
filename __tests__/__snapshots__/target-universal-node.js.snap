// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Target: Universal: Node Imports alias 1`] = `
"import { Button } from \\"../../src/components/button/Button\\";
import { logx } from \\"../../src/testonly\\";
import 'src/testonly';
logx(Button);"
`;

exports[`Target: Universal: Node Imports alias.tsx 1`] = `
"import { Button } from \\"../../src/components/button/Button\\";
import { logx } from \\"../../src/testonly\\";
import 'src/testonly';
logx(Button);"
`;

exports[`Target: Universal: Node Imports comments 1`] = `
"import _camelCase from \\"lodash/camelCase\\";

function translate(value) {
  return _camelCase(value);
}"
`;

exports[`Target: Universal: Node Imports deadcode 1`] = `
"var STATIC = false;
before();
after();"
`;

exports[`Target: Universal: Node Imports es2015: classes 1`] = `
"import _inheritsLoose from \\"@babel/runtime/helpers/esm/inheritsLoose\\";
export var Component =
/*#__PURE__*/
function (_Other) {
  _inheritsLoose(Component, _Other);

  function Component() {
    var _this = _Other.call(this) || this;

    _this.flag = true;
    return _this;
  }

  var _proto = Component.prototype;

  _proto.main = function main() {
    alert(this.foo);
  };

  return Component;
}(Other);"
`;

exports[`Target: Universal: Node Imports es2015: for_off 1`] = `
"import \\"core-js/modules/es.symbol\\";
import \\"core-js/modules/es.symbol.description\\";
import \\"core-js/modules/es.symbol.iterator\\";
import \\"core-js/modules/es.array.is-array\\";
import \\"core-js/modules/es.array.iterator\\";
import \\"core-js/modules/es.object.to-string\\";
import \\"core-js/modules/es.string.iterator\\";
import \\"core-js/modules/web.dom-collections.iterator\\";

function test() {
  for (var _iterator = tasks, _isArray = Array.isArray(_iterator), _i = 0, _iterator = _isArray ? _iterator : _iterator[Symbol.iterator]();;) {
    var _ref;

    if (_isArray) {
      if (_i >= _iterator.length) break;
      _ref = _iterator[_i++];
    } else {
      _i = _iterator.next();
      if (_i.done) break;
      _ref = _i.value;
    }

    try {
      executeCommands();
    } catch (error) {
      console.error(\\"Error\\");
    }
  }
}"
`;

exports[`Target: Universal: Node Imports es2015: generator 1`] = `
"function* idMaker() {
  var index = 0;

  while (true) {
    yield index++;
  }
}"
`;

exports[`Target: Universal: Node Imports es2015: set 1`] = `
"import \\"core-js/modules/es.array.iterator\\";
import \\"core-js/modules/es.object.to-string\\";
import \\"core-js/modules/es.set\\";
import \\"core-js/modules/es.string.iterator\\";
import \\"core-js/modules/web.dom-collections.iterator\\";
new Set([1, 2, 3]);"
`;

exports[`Target: Universal: Node Imports es2016: exponentiation 1`] = `"var result = Math.pow(2, 3);"`;

exports[`Target: Universal: Node Imports esnext: async 1`] = `
"import \\"core-js/modules/es.object.to-string\\";
import \\"core-js/modules/es.promise\\";

function wait() {
  return new Promise(function ($return, $error) {
    return Promise.resolve(loadData()).then(function () {
      try {
        return $return();
      } catch ($boundEx) {
        return $error($boundEx);
      }
    }, $error);
  });
}"
`;

exports[`Target: Universal: Node Imports esnext: async_loop 1`] = `
"import \\"core-js/modules/es.symbol\\";
import \\"core-js/modules/es.symbol.description\\";
import \\"core-js/modules/es.symbol.iterator\\";
import \\"core-js/modules/es.array.is-array\\";
import \\"core-js/modules/es.array.iterator\\";
import \\"core-js/modules/es.object.to-string\\";
import \\"core-js/modules/es.promise\\";
import \\"core-js/modules/es.string.iterator\\";
import \\"core-js/modules/web.dom-collections.iterator\\";

function executeTasks(tasks) {
  return new Promise(function ($return, $error) {
    var _iterator, _isArray, _i, _ref;

    _iterator = tasks, _isArray = Array.isArray(_iterator), _i = 0, _iterator = _isArray ? _iterator : _iterator[Symbol.iterator]();
    var $Loop_2_trampoline;

    function $Loop_2() {
      if (_isArray) {
        if (_i >= _iterator.length) return [1];
        _ref = _iterator[_i++];
      } else {
        _i = _iterator.next();
        if (_i.done) return [1];
        _ref = _i.value;
      }

      var $Try_1_Post = function () {
        try {
          return $Loop_2;
        } catch ($boundEx) {
          return $error($boundEx);
        }
      },
          $Try_1_Catch = function () {
        try {
          console.error(\\"Error\\");
          return $Try_1_Post();
        } catch ($boundEx) {
          return $error($boundEx);
        }
      };

      try {
        return Promise.resolve(executeCommands()).then(function () {
          try {
            return $Try_1_Post();
          } catch ($boundEx) {
            return $Try_1_Catch($boundEx);
          }
        }, $Try_1_Catch);
      } catch (error) {
        $Try_1_Catch(error)
      }
    }

    return ($Loop_2_trampoline = function (q) {
      while (q) {
        if (q.then) return void q.then($Loop_2_trampoline, $error);

        try {
          if (q.pop) {
            if (q.length) return q.pop() ? $Loop_2_exit.call(this) : q;else q = $Loop_2;
          } else q = q.call(this);
        } catch (_exception) {
          return $error(_exception);
        }
      }
    }.bind(this))($Loop_2);

    function $Loop_2_exit() {
      return $return();
    }
  });
}"
`;

exports[`Target: Universal: Node Imports esnext: catch_bind 1`] = `
"try {
  throw 0;
} catch (_unused) {
  doSomethingWhichDoesntCareAboutTheValueThrown();
} finally {
  doSomeCleanup();
}"
`;

exports[`Target: Universal: Node Imports esnext: class properties 1`] = `
"export var Component = function Component() {
  this.onClick = function () {};
};"
`;

exports[`Target: Universal: Node Imports esnext: decorators 1`] = `
"var _class,
    MyClass = annotation(_class = function MyClass() {}) || _class;

function annotation(target) {
  target.annotated = true;
}"
`;

exports[`Target: Universal: Node Imports esnext: dynamic import 1`] = `
"import \\"core-js/modules/es.object.to-string\\";
import \\"core-js/modules/es.promise\\";
import _interopRequireWildcard from \\"@babel/runtime/helpers/esm/interopRequireWildcard\\";
Promise.resolve().then(function () {
  return _interopRequireWildcard(require(\\"lodash\\"));
}).then(function (lodash) {
  console.log(lodash.camelCase(\\"hello world\\"));
});"
`;

exports[`Target: Universal: Node Imports esnext: object spread 1`] = `
"import \\"core-js/modules/es.object.assign\\";
var original = {
  bar: 10
},
    variant = Object.assign({
  foo: 1
}, original);"
`;

exports[`Target: Universal: Node Imports loadable 1`] = `
"import \\"core-js/modules/es.object.to-string\\";
import \\"core-js/modules/es.promise\\";
import _interopRequireWildcard from \\"@babel/runtime/helpers/esm/interopRequireWildcard\\";
import loadable from '@loadable/component';
export var Home = loadable({
  chunkName: function chunkName() {
    return \\"Home-JfPYC\\";
  },
  isReady: function isReady() {
    if (typeof __webpack_modules__ != 'undefined') {
      return !!__webpack_modules__[this.resolve(props)];
    }

    return false;
  },
  requireAsync: function requireAsync() {
    return Promise.resolve().then(function () {
      return _interopRequireWildcard(require(\\"./views/Home\\"));
    });
  },
  requireSync: function requireSync(props) {
    var id = this.resolve(props);

    if (typeof __webpack_require__ != 'undefined') {
      return __webpack_require__(id);
    }

    return eval('module.require')(id);
  },
  resolve: function resolve() {
    if (require.resolveWeak) {
      return require.resolveWeak(\\"./views/Home\\");
    }

    return eval('require.resolve')(\\"./views/Home\\");
  }
});
export var About = loadable({
  chunkName: function chunkName() {
    return \\"About-8yDxR\\";
  },
  isReady: function isReady() {
    if (typeof __webpack_modules__ != 'undefined') {
      return !!__webpack_modules__[this.resolve(props)];
    }

    return false;
  },
  requireAsync: function requireAsync() {
    return Promise.resolve().then(function () {
      return _interopRequireWildcard(require(\\"./views/About\\"));
    });
  },
  requireSync: function requireSync(props) {
    var id = this.resolve(props);

    if (typeof __webpack_require__ != 'undefined') {
      return __webpack_require__(id);
    }

    return eval('module.require')(id);
  },
  resolve: function resolve() {
    if (require.resolveWeak) {
      return require.resolveWeak(\\"./views/About\\");
    }

    return eval('require.resolve')(\\"./views/About\\");
  }
});
export var Contact = loadable({
  chunkName: function chunkName() {
    return \\"Contact-AIf6m\\";
  },
  isReady: function isReady(props) {
    if (typeof __webpack_modules__ != 'undefined') {
      return !!__webpack_modules__[this.resolve(props)];
    }

    return false;
  },
  requireAsync: function requireAsync() {
    return Promise.resolve().then(function () {
      return _interopRequireWildcard(require(\\"./views/Contact\\"));
    });
  },
  requireSync: function requireSync(props) {
    var id = this.resolve(props);

    if (typeof __webpack_require__ != 'undefined') {
      return __webpack_require__(id);
    }

    return eval('module.require')(id);
  },
  resolve: function resolve() {
    if (require.resolveWeak) {
      return require.resolveWeak(\\"./views/Contact\\");
    }

    return eval('require.resolve')(\\"./views/Contact\\");
  }
});"
`;

exports[`Target: Universal: Node Imports loadable magic-comments 1`] = `
"import \\"core-js/modules/es.object.to-string\\";
import \\"core-js/modules/es.promise\\";
import _interopRequireWildcard from \\"@babel/runtime/helpers/esm/interopRequireWildcard\\";
import loadable from '@loadable/component';
export var Home = loadable({
  chunkName: function chunkName() {
    return \\"Home-JfPYC\\";
  },
  isReady: function isReady() {
    if (typeof __webpack_modules__ != 'undefined') {
      return !!__webpack_modules__[this.resolve(props)];
    }

    return false;
  },
  requireAsync: function requireAsync() {
    return Promise.resolve().then(function () {
      return _interopRequireWildcard(require(\\"./views/Home\\"));
    });
  },
  requireSync: function requireSync(props) {
    var id = this.resolve(props);

    if (typeof __webpack_require__ != 'undefined') {
      return __webpack_require__(id);
    }

    return eval('module.require')(id);
  },
  resolve: function resolve() {
    if (require.resolveWeak) {
      return require.resolveWeak(\\"./views/Home\\");
    }

    return eval('require.resolve')(\\"./views/Home\\");
  }
});
export var About = loadable({
  chunkName: function chunkName() {
    return \\"About\\";
  },
  isReady: function isReady() {
    if (typeof __webpack_modules__ != 'undefined') {
      return !!__webpack_modules__[this.resolve(props)];
    }

    return false;
  },
  requireAsync: function requireAsync() {
    return Promise.resolve().then(function () {
      return _interopRequireWildcard(require(\\"./views/About\\"));
    });
  },
  requireSync: function requireSync(props) {
    var id = this.resolve(props);

    if (typeof __webpack_require__ != 'undefined') {
      return __webpack_require__(id);
    }

    return eval('module.require')(id);
  },
  resolve: function resolve() {
    if (require.resolveWeak) {
      return require.resolveWeak(\\"./views/About\\");
    }

    return eval('require.resolve')(\\"./views/About\\");
  }
});
export var Contact = loadable({
  chunkName: function chunkName() {
    return \\"Contact-AIf6m\\";
  },
  isReady: function isReady(props) {
    if (typeof __webpack_modules__ != 'undefined') {
      return !!__webpack_modules__[this.resolve(props)];
    }

    return false;
  },
  requireAsync: function requireAsync() {
    return Promise.resolve().then(function () {
      return _interopRequireWildcard(require(\\"./views/Contact\\"));
    });
  },
  requireSync: function requireSync(props) {
    var id = this.resolve(props);

    if (typeof __webpack_require__ != 'undefined') {
      return __webpack_require__(id);
    }

    return eval('module.require')(id);
  },
  resolve: function resolve() {
    if (require.resolveWeak) {
      return require.resolveWeak(\\"./views/Contact\\");
    }

    return eval('require.resolve')(\\"./views/Contact\\");
  }
});"
`;

exports[`Target: Universal: Node Imports loadable wildcard 1`] = `
"import \\"core-js/modules/es.object.to-string\\";
import \\"core-js/modules/es.promise\\";
import \\"core-js/modules/es.regexp.exec\\";
import \\"core-js/modules/es.string.replace\\";
import _interopRequireWildcard from \\"@babel/runtime/helpers/esm/interopRequireWildcard\\";
import loadable from '@loadable/component';
var page = \\"About\\",
    lang = \\"de_DE\\";
export var views = loadable({
  chunkName: function chunkName() {
    return (\\"views-\\" + page).replace(/[^a-zA-Z0-9_!§$()=\\\\-^°]+/g, \\"-\\");
  },
  isReady: function isReady() {
    if (typeof __webpack_modules__ != 'undefined') {
      return !!__webpack_modules__[this.resolve(props)];
    }

    return false;
  },
  requireAsync: function requireAsync() {
    return Promise.resolve().then(function () {
      return _interopRequireWildcard(require(\\"./views/\\" + page));
    });
  },
  requireSync: function requireSync(props) {
    var id = this.resolve(props);

    if (typeof __webpack_require__ != 'undefined') {
      return __webpack_require__(id);
    }

    return eval('module.require')(id);
  },
  resolve: function resolve() {
    if (require.resolveWeak) {
      return require.resolveWeak(\\"./views/\\" + page);
    }

    return eval('require.resolve')(\\"./views/\\" + page);
  }
});
export var leanIntlData = loadable({
  chunkName: function chunkName() {
    return (\\"lean-intl-locale-data-\\" + lang).replace(/[^a-zA-Z0-9_!§$()=\\\\-^°]+/g, \\"-\\");
  },
  isReady: function isReady() {
    if (typeof __webpack_modules__ != 'undefined') {
      return !!__webpack_modules__[this.resolve(props)];
    }

    return false;
  },
  requireAsync: function requireAsync() {
    return Promise.resolve().then(function () {
      return _interopRequireWildcard(require(\\"./lean-intl/locale-data/\\" + lang));
    });
  },
  requireSync: function requireSync(props) {
    var id = this.resolve(props);

    if (typeof __webpack_require__ != 'undefined') {
      return __webpack_require__(id);
    }

    return eval('module.require')(id);
  },
  resolve: function resolve() {
    if (require.resolveWeak) {
      return require.resolveWeak(\\"./lean-intl/locale-data/\\" + lang);
    }

    return eval('require.resolve')(\\"./lean-intl/locale-data/\\" + lang);
  }
});
export var reactIntlData = loadable({
  chunkName: function chunkName() {
    return (\\"react-intl-locale-data-\\" + lang).replace(/[^a-zA-Z0-9_!§$()=\\\\-^°]+/g, \\"-\\");
  },
  isReady: function isReady(props) {
    if (typeof __webpack_modules__ != 'undefined') {
      return !!__webpack_modules__[this.resolve(props)];
    }

    return false;
  },
  requireAsync: function requireAsync() {
    return Promise.resolve().then(function () {
      return _interopRequireWildcard(require(\\"./react-intl/locale-data/\\" + lang));
    });
  },
  requireSync: function requireSync(props) {
    var id = this.resolve(props);

    if (typeof __webpack_require__ != 'undefined') {
      return __webpack_require__(id);
    }

    return eval('module.require')(id);
  },
  resolve: function resolve() {
    if (require.resolveWeak) {
      return require.resolveWeak(\\"./react-intl/locale-data/\\" + lang);
    }

    return eval('require.resolve')(\\"./react-intl/locale-data/\\" + lang);
  }
});"
`;

exports[`Target: Universal: Node Imports lodash: async 1`] = `
"import _parallel from \\"async/parallel\\";

_parallel([]);"
`;

exports[`Target: Universal: Node Imports lodash: lodash 1`] = `
"import _camelCase from \\"lodash/camelCase\\";

_camelCase(\\"hello world\\");"
`;

exports[`Target: Universal: Node Imports lodash: recompose 1`] = `
"import _compose from \\"recompose/compose\\";

_compose([]);"
`;

exports[`Target: Universal: Node Imports macro: gql 1`] = `
"var query = {
  kind: \\"Document\\",
  definitions: [{
    kind: \\"OperationDefinition\\",
    operation: \\"query\\",
    name: {
      kind: \\"Name\\",
      value: \\"User\\"
    },
    variableDefinitions: [],
    directives: [],
    selectionSet: {
      kind: \\"SelectionSet\\",
      selections: [{
        kind: \\"Field\\",
        name: {
          kind: \\"Name\\",
          value: \\"user\\"
        },
        arguments: [{
          kind: \\"Argument\\",
          name: {
            kind: \\"Name\\",
            value: \\"id\\"
          },
          value: {
            kind: \\"IntValue\\",
            value: \\"5\\"
          }
        }],
        directives: [],
        selectionSet: {
          kind: \\"SelectionSet\\",
          selections: [{
            kind: \\"Field\\",
            name: {
              kind: \\"Name\\",
              value: \\"lastName\\"
            },
            arguments: [],
            directives: []
          }, {
            kind: \\"FragmentSpread\\",
            name: {
              kind: \\"Name\\",
              value: \\"UserEntry1\\"
            },
            directives: []
          }]
        }
      }]
    }
  }],
  loc: {
    start: 0,
    end: 79,
    source: {
      body: \\"\\\\n  query User {\\\\n    user(id: 5) {\\\\n      lastName\\\\n      ...UserEntry1\\\\n    }\\\\n  }\\\\n\\",
      name: \\"GraphQL request\\",
      locationOffset: {
        line: 1,
        column: 1
      }
    }
  }
};"
`;

exports[`Target: Universal: Node Imports macro: preval 1`] = `"var one = 1;"`;

exports[`Target: Universal: Node Imports promise 1`] = `
"import \\"core-js/modules/es.object.to-string\\";
import \\"core-js/modules/es.promise\\";
export default function () {
  return new Promise(function (resolve) {
    return resolve(42);
  });
}"
`;

exports[`Target: Universal: Node Imports react: jsx 1`] = `
"function render() {
  return React.createElement(\\"h1\\", null, \\"Hello\\");
}"
`;

exports[`Target: Universal: Node Imports react: jsx spread 1`] = `
"import \\"core-js/modules/es.object.assign\\";

function render(props) {
  return React.createElement(\\"h1\\", Object.assign({
    className: \\"large\\"
  }, props), \\"Hello\\");
}"
`;

exports[`Target: Universal: Node Imports require context 1`] = `
"if (process.env.NODE_ENV === \\"test\\") {
  require('babel-plugin-require-context-hook/register')();
}

var loader = require.context(__dirname, false, /\\\\.js/);"
`;

exports[`Target: Universal: Node Imports typescript.tsx 1`] = `
"export var hasPackageConfig = function hasPackageConfig(name) {
  try {
    return !!getPackageConfig()[name];
  } catch (_) {
    return false;
  }
};"
`;
